ledger/complete/mtrie/forest.go

```diff
@@ -190,19 +190,36 @@ func (f *Forest) Read(r *ledger.TrieRead) ([]ledger.Value, error) {
 	return orderedValues, nil
 }
 
-// Update updates the Values for the registers and returns rootHash and error (if any).
+// Update updates the Values for the registers, adds updated tries to forest,
+// and returns rootHash and error (if any).
+// In case there are multiple updates to the same register, Update will persist
+// the latest written value.
+func (f *Forest) Update(u *ledger.TrieUpdate) (ledger.RootHash, error) {
+	t, err := f.NewTrie(u)
+	if err != nil {
+		return ledger.RootHash(hash.DummyHash), err
+	}
+
+	err = f.AddTrie(t)
```

Comment:
in the case that NewTrie is returning parent trie (len zero) we are going to add it.

Reply:
Yes because `AddTrie()` would be a noop for trie that is already in the forest:

https://github.com/onflow/flow-go/blob/2127e44c0c86cf6af86ae967e6eba50ab99262df/ledger/complete/mtrie/forest.go#L386-L388

Larger code block for more context:

https://github.com/onflow/flow-go/blob/2127e44c0c86cf6af86ae967e6eba50ab99262df/ledger/complete/mtrie/forest.go#L374-L395